src/nvim/os/pty_process_unix.c

```diff
@@ -145,8 +145,12 @@ void pty_process_teardown(Loop *loop)
   uv_signal_stop(&loop->children_watcher);
 }
 
-static void init_child(PtyProcess *ptyproc) FUNC_ATTR_NONNULL_ALL
+static void init_child(PtyProcess *ptyproc)
+  FUNC_ATTR_NONNULL_ALL  // FUNC_ATTR_NORETURN ?
 {
+  // New session and progress-group. #6530
+  setsid();
```

Comment:
Why a new session instead of a new process group? Would `setpgid(0,0);` also do the job?

Reply:
Isn't a new session customary in a terminal emulator?